tests/test_engine.py

```diff
@@ -102,6 +101,26 @@ def test_invalid_format(self):
         ]
         self.assertRaises(TaurusConfigError, lambda: self.obj.configure(configs))
 
+    def test_included_configs(self):
+        configs = [
+            RESOURCES_DIR + "yaml/included-level1.yml",
+        ]
+        self.obj.configure(configs)
+        self.assertTrue(self.obj.config["level1"])
+        self.assertTrue(self.obj.config["level2"])
+        self.assertTrue(self.obj.config["level3"])
+        self.assertListEqual(['included-level2.yml', 'included-level3.yml'], self.obj.config["included-configs"])
+
+    def test_included_configs_cycle(self):
+        configs = [
+            RESOURCES_DIR + "yaml/included-circular1.yml",
+        ]
+        self.obj.configure(configs)
+        self.assertTrue(self.obj.config["level1"])
+        self.assertTrue(self.obj.config["level2"])
+        self.assertListEqual(['included-circular2.yml', 'included-circular1.yml', 'included-circular2.yml'],
```

Comment:
`engine.config['included-configs']` having a repeated value of `included-circular2.yml` is a bit non-obvious, but we can live with that.

Reply:
I have decided to not complicate code to avoid this, the cyclic config is undesired anyway